company-tng.el

```diff
@@ -106,12 +106,15 @@ confirm the selection and finish the completion."
      (let ((ov (make-overlay (point) (point))))
        (setq company-tng--overlay ov)
        (overlay-put ov 'priority 2))
-     (advice-add 'company-select-next :before-until 'company-tng--allow-unselected)
-     (advice-add 'company-fill-propertize :filter-args 'company-tng--adjust-tooltip-highlight))
+     (setq company-selection-default nil)
+     (unless company-selection-changed
+       (setq company-selection company-selection-default))
```

Comment:
Do we need to do this here? I imagine setting it to the "correct" value in `company-update-candidates` should be enough.

Reply:
The problem is `company-update-candidates` is called before `(company-call-frontends 'show)`, so the `company-selection-default` is not set to correct value at that time.
How about I introduce new `pre-show` command for front end so it can be used to setup `company-selection-default`?

Reply:
Then it shouldn't be set in the `show` handler. Frontends are a poor vehicle for changing behavior anyway.

It should be set in `company-tng-configure-default`. All tng users seem to have to call it.

Or we could add `company-tng-mode`, and it would be configured there.

Reply:
Add `company-tng-mode`

Reply:
Its docstring now says it's a frontend..?

Reply:
Do you have suggestion?

Reply:
Regarding the docstring, rather than copying the one from `company-tng-frontend` I think it will be clearer if says something like  "This minor mode enables \`company-tng-frontend' ".
So that users won't try to add `company-tng-mode` as a frontend by mistake.

Reply:
Done

Reply:
Nikita gave one. I'll probably edit the docstring later, too.

More importantly, do we have a reason for `company-tng-mode` not to set up the bindings automatically, like `company-tng-configure-default` did? Do you edit the bindings later yourself?

IIUC, `company-tng-configure-default` is not functional by itself anymore. So it would be a good idea to make it an obsolete alias for `company-tng-mode`, so that users' configurations don't break right away after the upgrade.

Reply:
I made it obsolete since `0.9.14`, please update the version and docstring as you see fit.